keras/src/distillation/distiller.py

```diff
@@ -0,0 +1,555 @@
+import keras
+from keras.src import tree
+from keras.src.api_export import keras_export
+from keras.src.distillation.distillation_loss import _convert_loss_to_function
+from keras.src.models.model import Model
+from keras.src.saving import serialization_lib
+
+
+@keras_export("keras.distillation.Distiller")
+class Distiller(Model):
+    """Distillation model for transferring knowledge from teacher to student.
+
+    Knowledge distillation transfers knowledge from a large, complex model
+    (teacher) to a smaller, simpler model (student). The student learns
+    from both ground truth labels and the teacher's predictions, often
+    achieving better performance than training on labels alone.
+
+    Arguments:
+        teacher: A trained `keras.Model` that serves as the knowledge source.
+            The teacher model is frozen during distillation.
+        student: A `keras.Model` to be trained through distillation.
+        strategies: List of distillation strategies to apply. Can be a single
+            strategy or a list of strategies like `LogitsDistillation`,
```

Comment:
"strategy instances like `keras.distillation.LogitsDistillation()`..."

Reply:
There is an inconsistency in terminology, where these are sometimes referred to as losses and sometimes as strategies. Which is it? (Could be both: training_strategy or something)

We should make the arg name here consistent with the object class name.